### ISSUE TITLE
Pagination headers

### ISSUE THREAD

**Issue by gwincr11** [Shopify/pyactiveresource#17]
Hello,

I am curious about support for pagination headers such as these:

```
$ curl --include 'https://localhost:3000/movies?page=5'
HTTP/1.1 200 OK
Link: <http://localhost:3000/movies?page=1>; rel="first",
  <http://localhost:3000/movies?page=173>; rel="last",
  <http://localhost:3000/movies?page=6>; rel="next",
  <http://localhost:3000/movies?page=4>; rel="prev"
Total: 4321
Per-Page: 10
```

I don't think this exists today, but it would be nice to add next page bindings to the api. Any opinions?

**Comment by kevinhughes27** [comment]
We don't include headers with next page details - have a look at the index reference for one of our resources: https://help.shopify.com/api/reference/article#index

If you need to fetch all items using pagination write a loop that increments the page param until you don't get anything back. Adding this utility to our client libraries has been discussed so if you make something you'd like to contribute back please open a PR

**Comment by gwincr11** [comment]
I would be interested in working on this. In your opinion would this be a pointer to another active resource object for the next page, or some sort of iterable?

**Comment by kevinhughes27** [comment]
What I've done in ruby before is create a `ArticleFinder` class that implements the ruby enumerable class which lets you use any of the standard lib iteration techniques. So to a user of this utility class / function they just loop through all objects but internally it makes a new API call every so often as required to get more data.

Does that make sense?

**Comment by gwincr11** [comment]
Yes that makes sense, so to abstract this you would create a new active resource class, maybe collection, that inherits from active resource and know how to fetch the next page?

**Comment by gwincr11** [comment]
Is there anyway to get the headers from the previous call?

**Comment by kevinhughes27** [comment]
you could store them in a variable before making the next call. I don't think you need to do this though to solve your problem.

**Comment by gwincr11** [comment]
Thanks @kevinhughes27 I don't need to do this, but it prevents me from making un-needed calls. I guess I was somewhat unclear, how do I get the response object from the urllib caller so I can store it?

**Comment by kevinhughes27** [comment]
How does it prevent you from making un-needed calls? I am not sure how to reach through to urllib you'll have to deep dive the code if thats what you need.

**Comment by gwincr11** [comment]
If the API is following spec rfc988 then the headers send back the next page if present. So if you are on the last page you know not to call again.

I can dig through I was just wondering if the active resource call was saving the calls response as state somewhere?

**Comment by kevinhughes27** [comment]
Yeah sorry I am not familiar enough with the internals
